gateway/starknet.go

```diff
@@ -45,19 +45,17 @@ type GatewayFunctionCall struct {
 type FunctionCall types.FunctionCall
 
 func (f FunctionCall) MarshalJSON() ([]byte, error) {
-	return json.Marshal(f)
-	// output := map[string]interface{}{}
-	// output["contract_address"] = f.ContractAddress.String()
-	// if f.EntryPointSelector != "" {
-	// 	output["entry_point_selector"] = f.EntryPointSelector
-	// }
-	// calldata := []string{}
-	// for _, v := range f.Calldata {
-	// 	data, _ := big.NewInt(0).SetString(v, 0)
-	// 	calldata = append(calldata, data.Text(10))
-	// }
-	// output["calldata"] = calldata
-	// return json.Marshal(output)
+	output := map[string]interface{}{}
```

Comment:
Just curious, did json.Marshal(f) cause problems?

Reply:
Yes, multiple examples of the json.Marshal were causing Stack Overflows.